ext/opcache/Optimizer/zend_inference.c

```diff
@@ -2496,7 +2515,8 @@ static int zend_update_type_info(const zend_op_array *op_array,
 
 			if (opline->extended_value == ZEND_ASSIGN_DIM) {
 				if (opline->op1_type == IS_CV) {
-					orig = assign_dim_result_type(orig, OP2_INFO(), tmp, opline->op2_type);
+					orig = assign_dim_result_type(orig, OP2_INFO(), tmp, opline->op2_type, 
+						CRT_CONSTANT_EX(op_array, opline, opline->op2, ssa->rt_constants));
```

Comment:
These CRT_CONSTANT_EX accesses should be guarded by `opline->op2_type == IS_CONST`.

Reply:
See 79a14f5 and b0074a6